src/ConfCrypt/Validation.hs

```diff
@@ -52,29 +52,29 @@ parameterTypesMatchSchema key ConfCryptFile {parameters} =
     where
         decryptAndCompare Parameter {paramName, paramValue, paramType} =
             catchError (runRule paramType paramName =<< decryptValue key paramValue)
-                       (const $ tell ["Error: Could not decrypt " <> paramName])
+                       (pure $ liftIO $ putStrLn ("Error: Could not decrypt " <> T.unpack paramName))
```

Comment:
Rather than `putStrLn`, this should accumulate errors. By adding the `MonadIO` constraint and using `putStrLn`, it becomes much more difficult to test this function.

Reply:
Errors are now accumulated in last commit.